### ISSUE TITLE
Support for Style/RegexpLiteral autocorrect when AllowInnerSlashes: false

### ISSUE THREAD

**Issue by tnir** [rubocop-hq/rubocop#5514]
## Expected behavior

autocorrect of `Style/RegexpLiteral` cop would be better to support `AllowInnerSlashes: false`.

## Actual behavior

`Style/RegexpLiteral` supports its autocorrect only for `EnforcedStyle` (as of RuboCop 0.52.1).

This limitation has existed intentionally since autocorrect was introduced in #1721.

## Steps to reproduce the problem

1. Prepare the files:
```sh
$ echo -e "Style/RegexpLiteral:\n  EnforcedStyle: mixed" | tee .rubocop.yml; echo -e 'Regexp.new(/https?:\/\//)' | tee regexp.rb
Style/RegexpLiteral:
  EnforcedStyle: mixed
Regexp.new(/https?:\/\//)
```

2. Execute rubocop with autocorrection:
```sh
$ rubocop -a
Inspecting 1 file
C

Offenses:

regexp.rb:1:12: C: Style/RegexpLiteral: Use %r around regular expression.
Regexp.new(/https?:\/\//)
           ^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.6 x86_64-darwin17)
```
